_ml-commons-plugin/connectors.md

```diff
@@ -0,0 +1,394 @@
+- [NVIDIA Triton](https://developer.nvidia.com/triton-inference-server) allows you to host text embedding models on NVIDIA's high performance model serving technology. Power semantic search queries and vector generation ingest pipelines using [GPU-acceleration]({{site.url}}{{site.baseurl}}/ml-commons-plugin/gpu-acceleration).
+- [Cohere](https://cohere.com/) allows you to use data from OpenSearch to power Cohere's large language models.
+
+## Prerequisites
+
+If you are an admin deploying wanting to use an ML connector, make sure that target model of the connector has already been deployed on your chosen platform. Furthermore, make sure that you have access to send and receive data to the third-party API for your connector. 
+
+When access control is enabled on your third-party platform, you can enter your security settings using the `authorization` or `credential` settings inside the connector API.
+
+### Adding trusted endpoints
+
+To make sure that you configure connectors in OpenSearch, add the trusted endpoints into your cluster settings using the `plugins.ml_commons.trusted_connector_endpoints_regex` setting, as shown in the following example:
+
+```
+PUT /_cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.trusted_connector_endpoints_regex": [
+            "^https://runtime\\.sagemaker\\..*\\.amazonaws\\.com/.*$",
+            "^https://api\\.openai\\.com/.*$",
+            "^https://api\\.cohere\\.ai/.*$",
+            "^https://bedrock\\..*\\.amazonaws.com/.*$"
+        ]
+    }
+}
+```
+{% include copy-curl.html %}
+
+### Enabling ML nodes
+
+Connectors also require the use of dedicated ML nodes. To make sure you have ML nodes enabled, update the following cluster settings:
+
+```
+PUT /_cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.only_run_on_ml_node": true,
+        "plugins.ml_commons.native_memory_threshold": 100
+    }
+}
+```
+{% include copy-curl.html %}
+
+
+### Setting up connector access control
+
+To enable access control on the connector API, use the following cluster setting. When enabled, the `authorization` or `credential` options will be required in order to use the connector API.
+
+```
+PUT /_cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.connector_access_control_enabled": false
+    }
+}
+```
+{% include copy-curl.html %}
+
+If successful, OpenSearch returns the following response:
+
+```
+{
+  "acknowledged": true,
+  "persistent": {
+    "plugins": {
+      "ml_commons": {
+        "connector_access_control_enabled": "true"
+      }
+    }
+  },
+  "transient": {}
+}
+```
+
+## Creating a connector
+
+The connector create API, `/_plugins/_ml/connectors/_create`, creates connections to third-party ML tools. Using the `endpoint` parameter, you can connect to any supported ML tool using their specific API endpoint. For example, to connect to a ChatGPT completetion model, you can connect using the `api.openai.com` as shown in the following example:
```

Comment:
Should we add more rules and guidances on how to write a connector template, so user can follow the rules and create connectors with more flexibility?

Reply:
Its more an example than anything. The configuration options and examples help with flexibility and customization when building a connector.

That being said, is there a better example we should use?